### DISCUSSION TITLE
How to inject Request (Telegram Context) ?

### DISCUSSION THREAD

**Discussion by felinto-dev** [bukhalo/nestjs-telegraf#748] [Q&A]
Hey!

https://docs.nestjs.com/fundamentals/injection-scopes#request-provider

I am developing a plugin for nestjs-telegraf
(still in alpha)
https://github.com/felinto-dev/tg-menu

And I have an advanced use-case. I need to inject the Telegraf request in a class like is possible in HTTP. Check the link above.

How I could do that using nestjs-telegraf?

Example

Expected solution:
```ts
import { Injectable, Scope, Inject } from '@nestjs/common';
import { CONTEXT } from '@nestjs/graphql';

@Injectable({ scope: Scope.REQUEST })
export class CatsService {
  constructor(@Inject(CONTEXT) private context) {}
}
```

**Comment by evilsprut** [comment]
@felinto-dev I'm very glad someone is taking advantage of what I've done. Unfortunately right now I don't have the opportunity to deal with the module and I can't tell you how to solve your problem. If you need fixes, be sure to send PR or someone else can help.
